packages/algob/src/types.ts

```diff
@@ -385,14 +385,37 @@ export interface Checkpoint {
 export type ASADef = z.infer<typeof ASADefSchema>;
 export type ASADefs = z.infer<typeof ASADefsSchema>;
 
-export interface GrpTxnParams {
-  fromAccount: Account
-  toAccountAddr: string
-  lsig: LogicSig
+export type execParams = AlgoTransferParam | AssetTransferParam | SSCCallsParam;
+
+export interface Sign {
+  sign: "sk" | "lsig"
```

Comment:
These should be enum as well. Also, should we ask the user for this? let's say a user wants to transfer some Algo's from A to B, then it makes sense for him to pass `txntype.algo`, but should we also ask him for `Sign` (as it should be understood that it will be signed by the secret key)? Can we handle this internally?

Reply:
No, because what if,  user wants to send algo using delegated Lsig. (for example, user wants to send algo only if fee is lower than some threshold)
We cannot assume what user wants.

Reply:
User will pass the lsig if he is using delegated lsig. Take `assetTransfer` function for example - we sign the txn using secret key, but we don't ask the user about the signing authority (passing "sk" explicitly).

Reply:
yes, i understand, that's why i added a type, if user wants he can choose.
For `assetTransfer` user can also use delegated lsig.
There can be many examples.
This is the problem which was not covered earlier , we had to make different functions for each case. But in this function that problem is solved